### ISSUE TITLE
Efficient net is sensitve to interpolation

### ISSUE THREAD

**Issue by NingNanXin** [lukemelas/EfficientNet-PyTorch#285]
First, thanks to the author's work,  Efficient Net is very useful in my work. And I stumbled upon a problem that this net is very sensitive to interpolation. I think it's very interesting and open an issue to record it.

I use Efficient Net to train a classify task that has 8 classes. In my dataset code, using the following code to process image
```
transforms.Compose([
        transforms.Resize((256, 256), Image.BILINEAR),
        transforms.RandomHorizontalFlip(),
        transforms.RandomCrop((224, 224)),
        transforms.ToTensor(),
    
    ]),
```

Only 20 epochs, Efficient Net achieve a good result. And I use onnx to infer an image but the output is not right. The process code is here:
```
img = cv2.imread("/data/data2/ningnanxin/efficientnet/trt/KMHGC41C1CU180653.jpg")
img = cv2.resize(img, (224, 224), interpolation=cv2.INTER_LINEAR)
img = np.asarray(img, dtype=np.float32)
img = img.astype(np.float32)
img = img.transpose((2, 0, 1))
img = img / 255.0
img = np.expand_dims(img, axis=0)
img = img.reshape(1, 3, 224, 224)


img1 = Image.open("/data/data2/ningnanxin/efficientnet/trt/KMHGC41C1CU180653.jpg")
transform = transforms.Compose([
            transforms.Resize((224, 224), Image.LINEAR),
            transforms.ToTensor(),
            ])
img1 = transform(img1)
img1 = np.asarray(img1, dtype=np.float32)
img1 = np.expand_dims(img1, axis=0)
img1 = img1.reshape(1, 3, 224, 224)
```

The result is here:
![image](https://user-images.githubusercontent.com/56398622/117929771-c1ba9f00-b32f-11eb-967c-7f4ad54c8c43.png)
Up is img result, down is img1 result. When I save img and img1, I can't find any significant difference.
To verify this problem, I use .pth to test, the code is following
```
def infer(self):
    self.model.eval()
    with torch.no_grad():
        for i in self.img_list:
            img = Image.open(i).convert("RGB")
            img = img.resize((224, 224), Image.BILINEAR)
            img = self.transform(img).unsqueeze(0).cuda()
            output = self.model(img)

            img1 = cv2.imread(i)
            img1 = cv2.resize(img1, (224, 224), interpolation=cv2.INTER_LINEAR)
            img1 = np.asarray(img1, dtype=np.float32)
            img1 = img1.astype(np.float32)
            img1 = img1.transpose((2, 0, 1))
            img1 = img1 / 255.0
            img1 = torch.tensor(img1)
            img1 = img1.unsqueeze(0).cuda()
            output1 = self.model(img1)

            for idx in torch.topk(output, k=5).indices.squeeze(0).tolist():
                prob = torch.softmax(output, dim=1)[0, idx].item()
                print(idx, prob)

            print("\n")

            for idx in torch.topk(output1, k=5).indices.squeeze(0).tolist():
                prob = torch.softmax(output1, dim=1)[0, idx].item()
                print(idx, prob)
```
The result is here:
![image](https://user-images.githubusercontent.com/56398622/117930684-e2372900-b330-11eb-8dfc-5a9d05882569.png)
Up is output, down is output1.
So if you want to deploy Efficient Net, you should be careful about the process code. 
Also,  using TensorRT to deploy should change "efficientnet_pytorch/utils.py", line 244,  
``` class Conv2dStaticSamePadding(nn.Conv2d): ```  Because TensorRT can't support irregular padding
``` 
        if pad_h > 0 or pad_w > 0:
            self.static_padding = nn.ZeroPad2d((pad_w // 2, pad_w - pad_w // 2,
                                                pad_h // 2, pad_h - pad_h // 2))
            self.padding = (pad_h // 2, pad_w // 2)
        else:
            self.static_padding = nn.Identity()
            self.padding = (0, 0)


    def forward(self, x):

        # x = self.static_padding(x)
        x = F.conv2d(x, self.weight, self.bias, self.stride, self.padding, self.dilation, self.groups)
        return x
```
Last, Thanks Again!

**Comment by JackyWang-001** [comment]
> BILINEAR

PIL   image mode RGB
OPENCV image mode BGR
Different image mode leads to different result.
